frontend/src/components/pages/UserProfilePage.tsx

```diff
@@ -1,18 +1,162 @@
-import { Box } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { useParams, useHistory } from "react-router-dom";
+import {
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Text,
+  useStyleConfig,
+} from "@chakra-ui/react";
+import { useQuery, useMutation } from "@apollo/client";
+import Header from "../navigation/Header";
+import {
+  SoftDeleteUserResponse,
+  SOFT_DELETE_USER,
+} from "../../APIClients/mutations/UserMutations";
+import {
+  MANAGE_USERS_TABLE_DELETE_USER_BUTTON,
+  MANAGE_USERS_TABLE_DELETE_USER_CONFIRMATION,
+} from "../../utils/Copy";
+import ConfirmationModal from "../utils/ConfirmationModal";
+import { GET_USER, User } from "../../APIClients/queries/UserQueries";
 
 type UserProfilePageProps = {
-  userId: string | undefined;
+  userId: string;
 };
 
 const UserProfilePage = () => {
   const { userId } = useParams<UserProfilePageProps>();
 
+  const [fullName, setFullName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [role, setRole] = useState<string>("");
+  const [confirmDeleteUser, setConfirmDeleteUser] = useState(false);
+
+  /* eslint-disable */
+  const [approvedLanguagesTranslation, setApprovedLanguagesTranslation] =
+    useState<{ [name: string]: number }>();
+  const [approvedLanguagesReview, setApprovedLanguagesReview] =
+    useState<{ [name: string]: number }>();
+
+  const history = useHistory();
+
+  useQuery(GET_USER(parseInt(userId, 10)), {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data: { userById: User }) => {
+      const user = data.userById;
+      const translationLanguages = user?.approvedLanguagesTranslation
+        ? JSON.parse(user.approvedLanguagesTranslation.replace(/'/g, '"'))
+        : {};
+      const reviewLanguages = user?.approvedLanguagesReview
+        ? JSON.parse(user.approvedLanguagesReview.replace(/'/g, '"'))
+        : {};
+      setFullName(`${user?.firstName} ${user?.lastName}`);
+      setEmail(user?.email!);
+      setApprovedLanguagesTranslation(translationLanguages);
+      setApprovedLanguagesReview(reviewLanguages);
+      setRole(
+        `Translator ${
+          Object.keys(reviewLanguages).length > 0 ? "& Reviewer" : ""
+        }`,
+      );
+    },
+    onError: () => {
+      history.push("/404");
+    },
+  });
+
+  const [deleteUser] = useMutation<{
+    response: SoftDeleteUserResponse;
+  }>(SOFT_DELETE_USER);
+
+  const closeModal = () => {
+    setConfirmDeleteUser(false);
+  };
+
+  const openModal = () => {
+    setConfirmDeleteUser(true);
+  };
+
+  const callSoftDeleteUserMutation = async () => {
+    await deleteUser({
+      variables: {
+        id: userId,
+      },
+    });
+    closeModal();
+    history.push("/");
+  };
+
+  const filterStyle = useStyleConfig("Filter");
   return (
-    <Box textAlign="center">
-      <h1>User Profile for {userId} :)</h1>
-    </Box>
+    <Flex direction="column" height="100vh">
+      <Header />
+      <Flex direction="row" flex={1}>
```

Comment:
nit: Flex direction is row by default, so I don't think you need to specify here.

Reply:
i dont mind leaving the `direction` prop to be extra readable